ironfish/src/network/peerNetwork.ts

```diff
@@ -1008,6 +1008,23 @@ export class PeerNetwork {
       // Force lazy deserialization of the transaction as a first sanity check
       const transaction = this.chain.verifier.verifyNewTransaction(message.transaction)
 
+      // Next, check the spends are valid and not already in the chain
+      const spendVerificationResult = await this.chain.verifier.verifyTransactionSpends(
```

Comment:
It looks like this will return `VerificationResultReason.ERROR` (from `pasRoot`) if the spend size is greater than the current merkle tree size. I think this means that we are seeing a transaction before we're seeing the block of the note it was created in. I was running this and seeing a lot of these errors. Although there were a lot of double spends, this error was the majority. 

Does it make sense to ignore those transactions? They would not currently be valid for the node but if we eventually catch up to the block they were created from they could become valid. Maybe we want those in the mempool. For compact blocks this would help especially